### PR TITLE
[ADD] sequence_reset_period (11.0)

### PR DESCRIPTION
**Pull request by etobella**
This module allows to configure when the period related sequences generates the new periods.

Initially, odoo generates them yearly, but sometimes it could be useful to configure them daily or monthly.

### CODE REVIEW

File: sequence_reset_period/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
```
Comment (JordiBForgeFlow) [inline sequence_reset_period/__init__.py]: Not needed in init files.

File: sequence_reset_period/models/ir_sequence.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+from odoo import fields, models
+from datetime import datetime, timedelta
+from dateutil.relativedelta import relativedelta
+
+
+class IrSequence(models.Model):
+    _inherit = "ir.sequence"
+
+    range_reset = fields.Selection([
```
Comment (JordiBForgeFlow) [inline sequence_reset_period/models/ir_sequence.py:12]: default to yearly?

File: sequence_reset_period/models/ir_sequence.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            date_to = date_from + relativedelta(months=1)
+            date_to += relativedelta(days=-1)
+        elif self.range_reset == 'yearly':
+            date_from -= timedelta(days=date_from.day-1)
+            date_from -= relativedelta(months=date_from.month - 1)
+            date_to = date_from + relativedelta(months=12)
+            date_to += relativedelta(days=-1)
+        return date_from.strftime('%Y-%m-%d'), date_to.strftime('%Y-%m-%d')
+
+    def _create_date_range_seq(self, date):
```
Comment (JordiBForgeFlow) [inline sequence_reset_period/models/ir_sequence.py]: ensure_one()

File: sequence_reset_period/models/ir_sequence.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    _inherit = "ir.sequence"
+
+    range_reset = fields.Selection([
+        ('daily', 'Daily'),
+        ('weekly', 'Weekly'),
+        ('monthly', 'Monthly'),
+        ('yearly', 'Yearly')
+    ])
+
+    def _compute_date_from_to(self, date):
```
Comment (JordiBForgeFlow) [inline sequence_reset_period/models/ir_sequence.py:19]: ensure_one()

Comment (JordiBForgeFlow) [reply sequence_reset_period/models/ir_sequence.py:19]: Perhaps you should consider, setting by default (pseudo code):
from dateutil.relativedelta import relativedelta
dt = datetime.strptime(day, '%d/%b/%Y')
if yearly:
start = date(dt.today().year, 1, 1)
end = date(dt.today().year, 12, 31)
elif 'weekly':
start = dt - timedelta(days=dt.weekday())
end = start + timedelta(days=6)
elif 'monthly':
start = dt + relativedelta(day=1)
end = dt + relativedelta(day=1, months=+1, days=-1)

**Review by JordiBForgeFlow — CHANGES_REQUESTED** [review]
See my comments

File: sequence_reset_period/models/ir_sequence.py (hunk 2 above)
Comment (etobella) [reply sequence_reset_period/models/ir_sequence.py:12]: I don't think so, because it shouldn't be set if `use_date_range` is False

**Review by JordiBForgeFlow — APPROVED** [review]

**Comment by JordiBForgeFlow** [comment]
:+1: Functional and technical review. Works like a charm!

File: sequence_reset_period/README.rst [hunk 5]
```diff
@@ -0,0 +1,38 @@
+.. image:: https://img.shields.io/badge/licence-LGPL--3-blue.svg
```
Comment (HviorForgeFlow) [inline sequence_reset_period/README.rst]: Better to use .png format

**Review by HviorForgeFlow — APPROVED** [review]
LGTM :+1: Functional and code review

**Pull request merged**